src/doom/p_inter.c

```diff
@@ -79,10 +80,12 @@ P_GiveAmmo
     if ( player->ammo[ammo] == player->maxammo[ammo]  )
 	return false;
 		
-    if (num)
```

Comment:
I think the `if (num)` condition may remain, even if it won't be triggered anymore during usual gameplay (I am curently looking for this patch to become least-invasive).

Also, could you probably move the `if (dropped)` condition behind the `gameskill` check?

Reply:
The last commit restores if (num) and wraps the new logic in it.

A negative 'num' is interpreted as "half the value of abs(num) clips". I don't *think* there's a reason to use this function for taking clips worth of ammo.

Still, if the 'dropped' parameter makes more sense, I can go back to that.

Reply:
> Still, if the 'dropped' parameter makes more sense, I can go back to that.

Yes, please do so. Sorry for being so unsteady, but you changed the code faster than I could properly review it.